### DISCUSSION TITLE
Some Confusion About n_one In FHDDM

### DISCUSSION THREAD

**Discussion by 9sea** [online-ml/river#1532] [Q&A]
The variable n_one is in river.drift.binary.fhddm.
My personal understanding is that a correct prediction is typically marked as 1, while an incorrect one is marked as 0, which is also mentioned in the original paper. However, in the code, it seems that a correct prediction is marked as 0 and an incorrect prediction as 1, which is the opposite of what I expected. This has caused me some confusion, and I would appreciate it if someone could clarify this for me. Thank you!

**Comment by smastelini** [comment]
Hi @9sea, following the same patterns as those used in the reference MOA implementation, the inputs refer to a 0-1 loss. So, 0 means no error and 1 represents a misclassification.

**Reply by 9sea** [reply]
u = self.n_one / self.sliding_window_size, u means the $p^1_{t}$ in original paper, i.e. the accuracy in sliding window. then trigger drift if self._u_max - u > self._epsilon. In another words, too much drop in accuracy is what triggers drift.

**Reply by 9sea** [reply]
![20240423224603](https://github.com/online-ml/river/assets/62071392/6e676130-d02c-4cc7-b921-99beb9d15d3d)
Hi, @smastelini , I did not find fhddm in MOA, but  I find another realization as picture shown.

**Reply by smastelini** [reply]
My bad, I missed the "F" at the beginning of the method name. I am pinging @gabrieljaguiar who did the implementation.

**Reply by 9sea** [reply]
I modify "update" function, Insert the line “x = 0 if x else 1” at the first line of the function， then I do some easy experiments, it has better performance. 
Thank you for your answer!

**Reply by gabrieljaguiar** [reply]
@9sea n_one is implemented as the number of correct predictions.

**Reply by 9sea** [reply]
@gabrieljaguiar The parameter x of the function update is 1, indicating a prediction error, and 0 indicating a correct prediction.
